lapack/testlapack/dlasy2.go

```diff
@@ -38,68 +42,83 @@ func Dlasy2Test(t *testing.T, impl Dlasy2er) {
+			for j := 0; j < n2; j++ {
+				b.Data[i*b.Stride+j] *= bignum
+			}
+		}
+	}
+
+	tlCopy := cloneGeneral(tl)
+	trCopy := cloneGeneral(tr)
+	bCopy := cloneGeneral(b)
 
 	scale, xnorm, ok := impl.Dlasy2(tranl, tranr, isgn, n1, n2, tl.Data, tl.Stride, tr.Data, tr.Stride, b.Data, b.Stride, x.Data, x.Stride)
-	if scale > 1 {
-		t.Errorf("invalid value of scale, want <= 1, got %v", scale)
+
+	// Check any invalid modifications in read-only input.
+	if !equalGeneral(tl, tlCopy) {
+		t.Errorf("%v: unexpected modification in TL", name)
 	}
-	if n1 == 0 || n2 == 0 {
-		return
+	if !equalGeneral(tr, trCopy) {
+		t.Errorf("%v: unexpected modification in TR", name)
+	}
+	if !equalGeneral(b, bCopy) {
+		t.Errorf("%v: unexpected modification in B", name)
 	}
-
-	prefix := fmt.Sprintf("Case n1=%v, n2=%v, isgn=%v", n1, n2, isgn)
 
 	// Check any invalid modifications of x.
 	if !generalOutsideAllNaN(x) {
-		t.Errorf("%v: out-of-range write to x\n%v", prefix, x.Data)
+		t.Errorf("%v: out-of-range write to x\n%v", name, x.Data)
 	}
 
-	var xnormWant float64
-	for i := 0; i < n1; i++ {
-		var rowsum float64
-		for j := 0; j < n2; j++ {
-			rowsum += math.Abs(x.Data[i*x.Stride+j])
-		}
-		if rowsum > xnormWant {
-			xnormWant = rowsum
-		}
+	if n1 == 0 || n2 == 0 {
+		return
 	}
+
+	if scale <= 0 || 1 < scale {
+		t.Errorf("%v: invalid value of scale, want in (0,1], got %v", name, scale)
+	}
+
+	xnormWant := dlange(lapack.MaxRowSum, x.Rows, x.Cols, x.Data, x.Stride)
 	if xnormWant != xnorm {
-		t.Errorf("%v: unexpected xnorm: want %v, got %v", prefix, xnormWant, xnorm)
+		t.Errorf("%v: unexpected xnorm: want %v, got %v", name, xnormWant, xnorm)
 	}
 
-	// Multiply b by scale to get the wanted right-hand side.
-	for i := 0; i < n1; i++ {
-		for j := 0; j < n2; j++ {
-			b.Data[i*b.Stride+j] *= scale
-		}
+	if !ok {
+		t.Logf("%v: Dlasy2 returned ok=false", name)
+		return
 	}
-	// Compute the wanted left-hand side.
-	lhsWant := randomGeneral(n1, n2, n2, rnd)
+
+	// Compute diff := op(TL)*X + sgn*X*op(TR) - scale*B.
+	diff := zeros(n1, n2, n2)
 	if tranl {
-		blas64.Gemm(blas.Trans, blas.NoTrans, 1, tl, x, 0, lhsWant)
+		blas64.Gemm(blas.Trans, blas.NoTrans, 1, tl, x, 0, diff)
 	} else {
-		blas64.Gemm(blas.NoTrans, blas.NoTrans, 1, tl, x, 0, lhsWant)
+		blas64.Gemm(blas.NoTrans, blas.NoTrans, 1, tl, x, 0, diff)
 	}
 	if tranr {
-		blas64.Gemm(blas.NoTrans, blas.Trans, float64(isgn), x, tr, 1, lhsWant)
+		blas64.Gemm(blas.NoTrans, blas.Trans, float64(isgn), x, tr, 1, diff)
 	} else {
-		blas64.Gemm(blas.NoTrans, blas.NoTrans, float64(isgn), x, tr, 1, lhsWant)
+		blas64.Gemm(blas.NoTrans, blas.NoTrans, float64(isgn), x, tr, 1, diff)
 	}
-	// Compare them.
 	for i := 0; i < n1; i++ {
 		for j := 0; j < n2; j++ {
-			diff := lhsWant.Data[i*lhsWant.Stride+j] - b.Data[i*b.Stride+j]
-			if math.Abs(diff) > tol && ok {
-				t.Errorf("%v: unexpected result, diff[%v,%v]=%v", prefix, i, j, diff)
-			}
+			diff.Data[i*diff.Stride+j] -= scale * b.Data[i*b.Stride+j]
```

Comment:
Should `scale * b.Data[i*b.Stride+j]` be inside a `float64` cast to enable testing on ARM?

Reply:
I think this is OK here until we see that it's not. It passes on travis and locally on my builder.